controllers/falcon_container/injector.go

```diff
@@ -0,0 +1,317 @@
+		imagePullSecrets = append(imagePullSecrets, corev1.LocalObjectReference{Name: falconContainer.Spec.Injector.ImagePullSecretName})
+	}
+	resources := &corev1.ResourceRequirements{}
+	if falconContainer.Spec.Injector.Resources != nil {
+		resources = falconContainer.Spec.Injector.Resources
+	}
+	var replicas int32 = 1
+	var rootUid int64 = 0
+	runNonRoot := true
+	initRunAsNonRoot := false
+	volumes := []corev1.Volume{
+		{
+			Name: injectorTLSSecretName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: injectorTLSSecretName,
+				},
+			},
+		},
+		{
+			Name: falconVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	initContainers := []corev1.Container{}
+	if falconContainer.Spec.Injector.AzureConfigPath != "" {
+		initContainers = append(initContainers, corev1.Container{
+			Name:            initContainerName,
+			ImagePullPolicy: falconContainer.Spec.Injector.ImagePullPolicy,
+			Image:           imageUri,
+			Command: []string{
+				"bash",
+				"-c",
+				"cp /run/azure.json /tmp/CrowdStrike/; chmod a+r /tmp/CrowdStrike/azure.json",
+			},
+			SecurityContext: &corev1.SecurityContext{
+				RunAsUser:    &rootUid,
+				RunAsNonRoot: &initRunAsNonRoot,
+				Privileged:   &initRunAsNonRoot,
+			},
+			VolumeMounts: []corev1.VolumeMount{{
+				Name:      falconVolumeName,
+				MountPath: falconVolumePath,
+			}, {
+				Name:      azureVolumeName,
+				MountPath: azureVolumePath,
+				ReadOnly:  true,
+			}},
+		})
+		volumes = append(volumes, corev1.Volume{
+			Name: azureVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: falconContainer.Spec.Injector.AzureConfigPath,
+					Type: &hostPathFile,
+				},
+			}})
+	}
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      injectorName,
+			Namespace: r.Namespace(),
+			Labels:    FcLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: FcLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: FcLabels,
+				},
+				Spec: corev1.PodSpec{
+					Affinity: &corev1.Affinity{
+						NodeAffinity: &corev1.NodeAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+								NodeSelectorTerms: []corev1.NodeSelectorTerm{
+									{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      "kubernetes.io/os",
+												Operator: corev1.NodeSelectorOpIn,
+												Values:   []string{"linux"},
+											},
+											{
+												Key:      "node-role.kubernetes.io/master",
+												Operator: corev1.NodeSelectorOpDoesNotExist,
+											},
```

Comment:
`control-plane` and `infra` and others will need to be added starting here in some envs. This should be in the CRD as a way to add more matchExpressions.

Reply:
Will be addressed in a future PR.